File: code/algomatrix/Group 1/Array.py

class MyArray:
    def __init__(self):
        self.length = 0
        self.data = {}

    def get(self, index):
        return self.data[index]

    def push(self, item):
        self.data[self.length] = item
        self.length += 1

    def pop(self):
        last_item = self.data[self.length - 1]
        del self.data[self.length - 1]
        self.length -= 1
        return last_item

    def delete(self, index):
        deleted_item = self.data[index]
        for i in range(index, self.length - 1):
            self.data[i] = self.data[i + 1]
        del self.data[self.length - 1]
        self.length -= 1
        return deleted_item

    def insert(self, index, item):
        for i in range(self.length, index, -1):
            self.data[i] = self.data[i - 1]
        self.data[index] = item
        self.length += 1

    def display(self):
        arr = [self.data[i] for i in range(self.length)]
        print(arr)

File: code/futureforge/group-32/array.js
// i worked on two data structures ie Array & linked List
// chukwuemeka chinecherem innocent  20/csc/028


// Array
class CustomArray {
    constructor() {
        this.length = 0;
        this.data = {};
    }

    // Get element at index
    get(index) {
        return this.data[index];
    }

    // Add element at the end
    push(item) {
        this.data[this.length] = item;
        this.length++;
    }

    // Remove last element
    pop() {
        if (this.length === 0) return undefined;
        const lastItem = this.data[this.length - 1];
        delete this.data[this.length - 1];
        this.length--;
        return lastItem;
    }

    // Delete element at index
    delete(index) {
        const deletedItem = this.data[index];
        this.shiftItems(index);
        return deletedItem;
    }

    // Helper method to shift items to the left
    shiftItems(index) {
        for (let i = index; i < this.length - 1; i++) {
            this.data[i] = this.data[i + 1];
        }
        delete this.data[this.length - 1];
        this.length--;
    }
}

// ouput
const customArray = new CustomArray();
customArray.push(1);
customArray.push(2);
customArray.push(3);
console.log(customArray.get(1)); // Output: 2
console.log(customArray.pop()); // Output: 3
customArray.delete(0);
console.log(customArray); // Output: { length: 1, data: { '1': 2 } }

